packages/adapter-prisma/src/index.ts

```diff
@@ -25,53 +25,75 @@ import type {
 } from "@auth/core/adapters"
 
 export function PrismaAdapter(
-  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>
+  prisma: PrismaClient | ReturnType<PrismaClient["$extends"]>,
+  options?: {
+    accountModel?: string
+    authenticatorModel?: string
+    sessionModel?: string
+    userModel?: string
+    verificationTokenModel?: string
+  }
 ): Adapter {
   const p = prisma as PrismaClient
+  const accountModel =
+    p[(options?.accountModel ?? "account") as keyof PrismaClient]
+  const authenticatorModel =
+    p[(options?.authenticatorModel ?? "authenticator") as keyof PrismaClient]
+  const sessionModel =
+    p[(options?.sessionModel ?? "session") as keyof PrismaClient]
+  const userModel = p[(options?.userModel ?? "user") as keyof PrismaClient]
+  const verificationTokenModel =
+    p[
+      (options?.verificationTokenModel ??
+        "verificationToken") as keyof PrismaClient
+    ]
   return {
     // We need to let Prisma generate the ID because our default UUID is incompatible with MongoDB
-    createUser: ({ id, ...data }) => p.user.create(stripUndefined(data)),
-    getUser: (id) => p.user.findUnique({ where: { id } }),
-    getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
+    createUser: ({ id, ...data }) => userModel.create(stripUndefined(data)),
+    getUser: (id) => userModel.findUnique({ where: { id } }),
+    getUserByEmail: (email) => userModel.findUnique({ where: { email } }),
     async getUserByAccount(provider_providerAccountId) {
-      const account = await p.account.findUnique({
+      const account = await accountModel.findUnique({
         where: { provider_providerAccountId },
         select: { user: true },
       })
       return (account?.user as AdapterUser) ?? null
     },
     updateUser: ({ id, ...data }) =>
-      p.user.update({
+      userModel.update({
         where: { id },
         ...stripUndefined(data),
       }) as Promise<AdapterUser>,
     deleteUser: (id) =>
-      p.user.delete({ where: { id } }) as Promise<AdapterUser>,
+      userModel.delete({ where: { id } }) as Promise<AdapterUser>,
     linkAccount: (data) =>
-      p.account.create({ data }) as unknown as AdapterAccount,
+      accountModel.create({ data }) as unknown as AdapterAccount,
```

Comment:
```suggestion
      accountModel.create({ data }) as Promise<AdapterAccount>,
```
With the changes above, you can also do this change iirc